docs/DeveloperGuide.md

```diff
@@ -1,38 +1,323 @@
+
+#### Overview
+
+The abstract Command class has been implemented to introduce an additional layer of abstraction between I/O and command execution, allowing for separation of handling command keywords and executing commands.
+
+
+<u>Implementation Details</u>
+
+The following diagram is an inheritance diagram for Command and its children classes. 
+This has been heavily simplified and only shows the key commands.
+
+![CommandInheritance.png](diagrams%2Fimages%2FCommandInheritance.png)
+
+
+The following diagram is a sequence diagram for execution of Command.
+
+![CommandExecutionSequence.png](diagrams%2Fimages%2FCommandExecutionSequence.png)
+
+Commands interact with `UI` and `Parser` classes via `Main`, as illustrated in the following class diagram:
+
+![UiToCommand.png](diagrams%2Fimages%2FUiToCommand.png)
+
+### Storage
+
+<u>Overview</u>
+
+Majority of the storage handling occurs in the ExpenseList class. 
+This is to keep the storage tightly coupled with the expense data and ensures that when the expense list is updated,
+the storage format is updated accordingly.
+
+OpenCSV is used for the storage format to allow for reliable handling of CSV files. 
+The CSV file format was used to allow expenses to be easily exported/ imported to and from other programs (like Excel).
+
+In the future, if the state expands to be greater than an ExpenseList, it would be better for a separate class to be
+created for solely handling Storage. 
+
+### Expense and Expense List
+
+<u>Overview</u>
+
+The `Expense` class represents an individual expense with a price, description, category and the date added.
+
+The `ExpenseList` class manages a collection of `Expense` objects. 
+It allows for the addition, editing and deletion of expenses.
+
+<u>Methods</u>
+
+The `Expense` class has no notable methods.
+
+The `ExpenseList` class has the following key methods: 
 
-## Design & implementation
+|     Method      |            Description            |
+|:---------------:|:---------------------------------:|
+|  `addExpense`   |    Adds an expense to the list    |
+| `deleteExpense` | Removes an expense from the list  |
+|  `editExpense`  |   Edits an expense in the list    |
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+<u>Design Considerations</u>
 
+The setters in `Expense` class checks for null and blank.
+The `Expense` constructors also do, as they use those setters.
+
+The nontrivial methods in `ExpenseList` class contain some sort of exception handling. 
+
+<u>Implementation Details</u>
+
+UML class diagram to show the dependency between `Command` and `ExpenseList` classes:
+
+![CommandAndExpenseList.png](diagrams%2Fimages%2FCommandAndExpenseList.png)
+
+The following diagram is a UML class diagram for `Expense` and `ExpenseList`:
+
+![ExpenseAndExpenseList.png](diagrams%2Fimages%2FExpenseAndExpenseList.png "UML Class Diagram for Expense and ExpenseList")
+
+### Expense Filter
```

Comment:
would it be better to have a sequence diagram to showcase each of the commands and how they work